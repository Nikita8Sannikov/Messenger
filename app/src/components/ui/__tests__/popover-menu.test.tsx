import { fireEvent, render, screen } from "@testing-library/react";

import PopoverMenu from "@components/ui/popover-menu";

describe("PopoverMenu", () => {
	const mockAnchorEl = document.createElement("div");
	const mockOnClose = jest.fn();
	const mockItemClick = jest.fn();
    
	const mockProps = {
		id: "test-menu",
		anchorEl: mockAnchorEl,
		anchorOrigin: {
			vertical: "top" as const,
			horizontal: "left" as const,
		},
		open: true,
		onClose: mockOnClose,
		items: [
			{
				onClick: mockItemClick,
				title: "Test Item 1",
				className: "custom-class",
			},
			{
				onClick: mockItemClick,
				title: "Test Item 2",
				icon: <span>Icon</span>,
			},
		],
	};

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("should render menu items", () => {
		render(<PopoverMenu {...mockProps} />);
		expect(screen.getByText("Test Item 1")).toBeInTheDocument();
		expect(screen.getByText("Test Item 2")).toBeInTheDocument();
	});

	it("should render menu items with icons", () => {
		render(<PopoverMenu {...mockProps} />);
		expect(screen.getByText("Icon")).toBeInTheDocument();
	});

	it("should call onClick and onClose when menu item is clicked", () => {
		render(<PopoverMenu {...mockProps} />);
		fireEvent.click(screen.getByText("Test Item 1"));
		expect(mockItemClick).toHaveBeenCalled();
		expect(mockOnClose).toHaveBeenCalled();
	});

	it("should apply custom className to menu item", () => {
		render(<PopoverMenu {...mockProps} />);
		const menuItem = screen.getByText("Test Item 1").closest(".custom-class");
		expect(menuItem).toBeInTheDocument();
	});

	it("should not render when open is false", () => {
		render(<PopoverMenu {...mockProps} open={false} />);
		expect(screen.queryByText("Test Item 1")).not.toBeInTheDocument();
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<PopoverMenu {...mockProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
