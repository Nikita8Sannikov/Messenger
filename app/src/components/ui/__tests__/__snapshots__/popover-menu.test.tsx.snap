// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverMenu matches snapshot 1`] = `<DocumentFragment />`;
